src/main/java/org/nastya/service/UserClient/UserAuthorizationChecker.java

```diff
@@ -0,0 +1,26 @@
+package org.nastya.service.UserClient;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestTemplate;
+
+@Component
+public class UserAuthorizationChecker {
+    private static final Logger log = LoggerFactory.getLogger(UserAuthorizationChecker.class);
+    private final RestTemplate restTemplate;
+
+    public UserAuthorizationChecker(RestTemplate restTemplate) {
+        this.restTemplate = restTemplate;
+    }
+
+    public boolean isUserAuthorized(String sessionId) {
+        log.info("Checking authorization for sessionId: {}", sessionId);
+        String baseUrl = "http://localhost:8081/book-shop-authenticator";
```

Comment:
String baseUrl = "http://localhost:8081/book-shop-authenticator";
Has to be injected into the constructor as @Value field.  The value ""http://localhost:8081/book-shop-authenticator"" should come from application.yml file. Creat a custome field there.